File: leetcode/2011.执行操作后的变量值.cpp
#include <iostream>
#include <vector>
#include <unordered_map>
using namespace std;
/*
 * @lc app=leetcode.cn id=2011 lang=cpp
 *
 * [2011] 执行操作后的变量值
 */

// @lc code=start
class Solution
{
public:
    unordered_map<string, int> map = {{"X++", 1}, {"++X", 1}, {"--X", -1}, {"X--", -1}};

    int finalValueAfterOperations(vector<string> &operations)
    {
        int res = 0;
        for (auto &&op : operations)
        {
            res += map[op];
        }
        return res;
    }
};
// @lc code=end

File: leetcode/2011.执行操作后的变量值.py
#
# @lc app=leetcode.cn id=2011 lang=python3
#
# [2011] 执行操作后的变量值
#
from typing import List


# @lc code=start
class Solution:

    MAP = {"X++": 1, "++X": 1, "--X": -1, "X--": -1}

    def finalValueAfterOperations(self, operations: List[str]) -> int:
        return sum(self.MAP[op] for op in operations)
# @lc code=end
